pkg/utils/imagevector/imagevector.go

```diff
@@ -227,23 +251,57 @@ func checkConstraint(constraint, version *string) (score int, ok bool, err error
 	return score, true, nil
 }
 
+func checkArchitectureConstraint(source, desired *string) (score int, ok bool, err error) {
+	// if image doesn't have a architecture tag it is considered as multi arch image
+	// and if worker pool machine doesn't have architecture tag it is by default considered amd64 machine.
+	var sourceArch, desiredArch = v1beta1constants.ArchitectureWildcard, v1beta1constants.ArchitectureAMD64
+
+	if source != nil {
+		sourceArch = *source
+	}
+	if desired != nil {
+		desiredArch = *desired
+	}
+
+	if desiredArch == v1beta1constants.ArchitectureWildcard {
+		return 0, false, fmt.Errorf("desired architecture cannot be '*'")
+	}
```

Comment:
can be dropped, as the shoot would already be validated to not specify the wildcard